tmf/org.eclipse.tracecompass.tmf.core/src/org/eclipse/tracecompass/tmf/core/statistics/TmfStateStatistics.java

```diff
@@ -117,40 +117,63 @@ public void dispose() {
         if (quark == ITmfStateSystem.INVALID_ATTRIBUTE) {
             return Collections.emptyList();
         }
-        List<Long> times = new ArrayList<>();
+        List<@NonNull Long> times = new ArrayList<>();
         for (int i = 0; i < timeRequested.length; i++) {
+            /* create padding for every window of time before the start time */
             if (timeRequested[i] < fTotalsStats.getStartTime()) {
-                times.add(fTotalsStats.getStartTime());
-            } else if (timeRequested[i] < fTotalsStats.getCurrentEndTime()) {
-                times.add(timeRequested[i]);
+                list.add(0L);
             } else {
+                if (!list.isEmpty()) {
+                    /* replace this bucket with the [start time, next time] */
+                    times.add(fTotalsStats.getStartTime());
+                    list.remove(list.size() - 1);
+                }
+                times.add(timeRequested[i]);
+            }
+            if (timeRequested[i] > fTotalsStats.getCurrentEndTime()) {
                 times.add(fTotalsStats.getCurrentEndTime());
                 break;
             }
         }
         try (FlowScopeLog log = new FlowScopeLogBuilder(LOGGER, Level.FINE, "StateStatistics:histogramQuery").build()) { //$NON-NLS-1$
+            addHistogramValuesFromQuery2d(quark, times, list);
```

Comment:
Opening the event densitiv view in Eclipse or when running the trace server, this method throws a ClassCastException which is swallowed. I added a try- catch to get the stack back trace. For that reason the histogram/Overview view is not populated with theia-trace-extension/vscode-trace-extension and event density view in Eclipse.
```
java.lang.ClassCastException: class java.lang.Integer cannot be cast to class java.lang.Long (java.lang.Integer and java.lang.Long are in module java.base of loader 'bootstrap')
	at org.eclipse.tracecompass.statesystem.core.interval.ITmfStateInterval.getValueLong(ITmfStateInterval.java:110)
	at org.eclipse.tracecompass.tmf.core.statistics.TmfStateStatistics.addHistogramValuesFromQuery2d(TmfStateStatistics.java:171)
	at org.eclipse.tracecompass.tmf.core.statistics.TmfStateStatistics.histogramQuery(TmfStateStatistics.java:140)
	at org.eclipse.tracecompass.internal.tmf.core.histogram.HistogramDataProvider.fetchXY(HistogramDataProvider.java:128)
	at org.eclipse.tracecompass.tmf.core.model.xy.TmfTreeXYCompositeDataProvider.getXyResponses(TmfTreeXYCompositeDataProvider.java:187)
	at org.eclipse.tracecompass.tmf.core.model.xy.TmfTreeXYCompositeDataProvider.fetchXY(TmfTreeXYCompositeDataProvider.java:136)
	at org.eclipse.tracecompass.incubator.internal.trace.server.jersey.rest.core.services.DataProviderService.getXY(DataProviderService.java:390)
```

Reply:
I modified the Statistics analysis to generate long state values instead of integers, so all statistics ht files will have to be regenerated. I did not observe this when I regenerate the ht files.

Reply:
You need to update the version number of the state provider of the statistics then. Otherwise, the state systems are not rebuilt for existing traces and users will get that issue.